### PR TITLE
feat: expiring todo comments

### PR DESCRIPTION
**Pull request by u2v22**
https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/expiring-todo-comments.md

Slack discussion for reference: https://neofinancial.slack.com/archives/C01GZFPQTFG/p1686003619708169

**_Summary_**:

After eslint 8.40.0 there's an error that shows up when upgrading eslint
image

=======================================================================================

**_Proposed solution_**

Adding this as a rule to our config.

Bumped to 0.8.0 because of the major version(s) in the unicorn plugin. 

Also ended up with (in mortgage-service):
- Unnecessarily cloning an array
- Unexpected negated condition
...both seemed like reasonable "errors"

Some examples (tested in mortgage-service):

Screenshot 2023-06-05 at 8 03 18 PM

Screenshot 2023-06-05 at 8 06 46 PM

image

Screenshot 2023-06-05 at 8 04 48 PM

Screenshot 2023-06-05 at 5 35 59 PM

Screenshot 2023-06-05 at 5 36 34 PM

Screenshot 2023-06-05 at 5 37 03 PM

### CODE REVIEW

File: config-base.js [hunk 1]
```diff
@@ -118,6 +118,12 @@ module.exports = {
 
     'unicorn/catch-error-name': 'off',
     'unicorn/consistent-function-scoping': 'off',
+    'unicorn/expiring-todo-comments': [
+      'warn',
+      {
+        terms: ['todo'],
```
Comment (u2v22) [inline config-base.js:124]: Case insensitive 👍

**Review by leah-is-offline — APPROVED** [review]
Not sure how I feel about adding dates to to dos, but I know that I feel better about it, then todo's going unnoticed forever 

also what happens if the letters "todo" are in a comment but it doesn't start with the work todo? not that theres very many of them https://www.worddb.com/words/containing/todo

**Review by neo-bryce-brandford — COMMENTED** [review]
may be a dumb question but does the warning stop the commit process or does it just log the warning to be fixed? if it is throwing an error are you able to change the date to a future date to clear the error or do you have to remove the todo?

would also be cool if we could find a way to git blame and add the GitHub user who added the todo.

**Comment by u2v22** [comment]
> ?

Yeah @neo-bryce-brandford the idea would be that this wouldn't stop someone from committing but would leave logs indicating the issue(s).

**Review by MaxGoo — APPROVED** [review]

**Comment by AdamTranquilla** [comment]
> Not sure how I feel about adding dates to to dos, but I know that I feel better about it, then todo's going unnoticed forever
> 
> also what happens if the letters "todo" are in a comment but it doesn't start with the work todo? not that theres very many of them [worddb.com/words/containing/todo](https://www.worddb.com/words/containing/todo)

How about [replicationDTOToDomain](https://neofinancial.sourcegraph.com/search?q=context%3Aglobal+replicationDTOToDomain&patternType=standard&sm=1&groupBy=repo) lol

**Pull request merged**